Dockerfile

```diff
@@ -18,21 +22,19 @@ RUN set -ex \
 # Copy in your requirements file
 ADD requirements.txt /requirements.txt
 
-# OR, if you’re using a directory for your requirements, copy everything (comment out the above and uncomment this if so):
+# OR, if you're using a directory for your requirements, copy everything (comment out the above and uncomment this if so):
 # ADD requirements /requirements
 
 # Install build deps, then run `pip install`, then remove unneeded build deps all in a single step.
 # Correct the path to your production requirements file, if needed.
 RUN set -ex \
     && BUILD_DEPS=" \
-        build-essential \
-        libpcre3-dev \
-        libpq-dev \
+    build-essential \
+    libpcre3-dev \
+    libpq-dev \
     " \
     && apt-get update && apt-get install -y --no-install-recommends $BUILD_DEPS \
-    && python3.7 -m venv /venv \
-    && /venv/bin/pip install -U pip \
-    && /venv/bin/pip install --no-cache-dir -r /requirements.txt \
+    && pip install --no-cache-dir -r /requirements.txt \
```

Comment:
Not using a venv means we will pick up any python packages installed system-wide. I'd feel more secure if we created our own venv and only put in what we knew we needed.

Reply:
It looks like these docker images (including the non-slim version) don't have any python packages installed system-wide:

```
$ docker run -it python:3.7-slim bash
root@244f278aeb78:/# pip freeze
root@244f278aeb78:/# python -m venv /venv
root@244f278aeb78:/# /venv/bin/pip freeze
root@244f278aeb78:/# 
```
```
$ docker run -it python:3.7 bash
root@6d616a71de0e:/# pip freeze
root@6d616a71de0e:/# python -m venv /venv
root@6d616a71de0e:/# /venv/bin/pip freeze
root@6d616a71de0e:/# 
```

Reply:
I'm just saying I'd rather not have to go verify that, or worry that it might change someday.  Using a venv seems like cheap insurance.

Reply:
I understand Dan's reluctance since I've been used to using venvs everywhere, but I think it's safe to make the assumption that the slim image will not have other packages installed. I was surprised that even the non "slim" version had no packages. It also seems pretty easy to verify as Tobias has shown, so I'm OK with this. We've been using this pattern on many of our production Docker images these days.